sw/host/opentitanlib/src/app/mod.rs

```diff
@@ -624,6 +624,27 @@ impl TransportWrapper {
         std::thread::sleep(reset_delay);
         Ok(())
     }
+
+    pub fn reset_target_w_strappings(
+        &self,
+        reset_delay: Duration,
+        clear_uart_rx: bool,
+        added_strappings: &[&str],
+        hold_added_strappings: bool,
+    ) -> Result<()> {
```

Comment:
I'm not a huge fan of lots of Booleans in APIs that change behaviour, do you have any ideas for making the `clear_uart_rx` and `hold_added_strappings` behaviour more explicit at the call-site?

Something like this feels Rusty to me:

```rust
target.apply_strappings(&["RMA_BOOTSTRAP"]);
target.reset(Duration::from_secs(2), UartRx::Clear)?;

// or

let held_straps = target.hold_straps(&["RMA_BOOTSTRAP", "PINMUX_TAP_LC"])?;
target.reset(Duration::from_secs(2), UartRx::Keep)?;
// `held_straps` is dropped and removes the strappings
```